plugin/pkg/scheduler/algorithm/predicates/predicates_test.go

```diff
@@ -571,10 +573,17 @@ func TestPodFitsHost(t *testing.T) {
 	}
 }
 
-func newPod(host string, hostPorts ...int) *v1.Pod {
+func newPod(host string, hostPortInfos ...string) *v1.Pod {
 	networkPorts := []v1.ContainerPort{}
-	for _, port := range hostPorts {
-		networkPorts = append(networkPorts, v1.ContainerPort{HostPort: int32(port)})
+	for _, portInfo := range hostPortInfos {
+		// assume portInfo is "protocol/hostip/hostport"
```

Comment:
You should not enforce `protocol/hostip/hostport` here and anywhere else, please consider use a helper function which can deal with any input port properly.

So in predicate, you can call `diff(helper(existingPort), helper(requestedPort))` to make decision.

Reply:
hmm, I once had the same concern with you.Then I realize if we don't use this "protocol/hostip/hostport" as a key, we will also have to change the existing function ```schedutil.GetUsedPorts(pod)``` and ``` existingPorts := nodeInfo.UsedPorts()```. They all return a map[string]bool for now.

Reply:
About your suggestion "diff(helper(existingPort), helper(requestedPort))", I wonder ```existingPort``` and ```requestedPort``` is string type or a struct type(containerPort)

Reply:
@k82cn PTAL waht's your opinion on "You should not enforce protocol/hostip/hostport here and anywhere else, please consider use a helper function which can deal with any input port properly.

So in predicate, you can call diff(helper(existingPort), helper(requestedPort)) to make decision."

Reply:
I'd like to +1 to helper functions :). maybe encode/decode helper functions to transform between str and trip values.

Reply:
I will define 
```
type hostPortInfo struct {
       hostIP string
       hostPort string 
       protocol string
}
```
we define some helper method(encode/decode and compare if exists ant conflict) on hostPortInfo to do the job, it's more OO.

Reply:
I plan to define these in utils.go under predicates package. WDYT